src/tests/ReplayBatchRendererTest.cpp

```diff
@@ -0,0 +1,195 @@
+  const std::string userPrefix = "sensor_";
+  const std::string screenshotPrefix = "ReplayBatchRendererTest_env";
+  const std::string screenshotExtension = ".png";
+
+  std::vector<std::string> serKeyframes;
+  for (int envIndex = 0; envIndex < numEnvs; envIndex++) {
+    SimulatorConfiguration simConfig{};
+    simConfig.activeSceneName = vangogh;
+    simConfig.enableGfxReplaySave = true;
+    simConfig.createRenderer = false;
+
+    auto sim = Simulator::create_unique(simConfig);
+
+    // add and pose objects
+    {
+      auto objAttrMgr = sim->getObjectAttributesManager();
+      objAttrMgr->loadAllJSONConfigsFromPath(
+          Cr::Utility::Path::join(TEST_ASSETS, "objects/donut"), true);
+
+      auto handles = objAttrMgr->getObjectHandlesBySubstring("donut");
+      CORRADE_VERIFY(!handles.empty());
+
+      auto rigidObj0 =
+          sim->getRigidObjectManager()->addBulletObjectByHandle(handles[0]);
+      rigidObj0->setTranslation(
+          Mn::Vector3(1.5f, 1.0f + envIndex * 0.2f, 0.7f));
+      rigidObj0->setRotation(
+          Mn::Quaternion::rotation(Mn::Deg(45.f - envIndex * 5.f),
+                                   Mn::Vector3(1.f, 0.f, 0.f).normalized()));
+
+      auto rigidObj1 =
+          sim->getRigidObjectManager()->addBulletObjectByHandle(handles[0]);
+      rigidObj1->setTranslation(
+          Mn::Vector3(1.5f, 1.2f + envIndex * 0.2f, -0.7f));
+      rigidObj1->setRotation(
+          Mn::Quaternion::rotation(Mn::Deg(30.f + envIndex * 5.f),
+                                   Mn::Vector3(0.f, 0.f, 1.f).normalized()));
+    }
+
+    auto& recorder = *sim->getGfxReplayManager()->getRecorder();
+    recorder.addUserTransformToKeyframe(
+        userPrefix + sensorName, Mn::Vector3(3.3f, 1.3f + envIndex * 0.1f, 0.f),
+        Mn::Quaternion::rotation(Mn::Deg(80.f + envIndex * 5.f),
+                                 Mn::Vector3(0.f, 1.f, 0.f)));
+
+    std::string serKeyframe = esp::gfx::replay::Recorder::keyframeToString(
+        recorder.extractKeyframe());
+    serKeyframes.emplace_back(std::move(serKeyframe));
+  }
+
+  std::vector<esp::sensor::SensorSpec::ptr> sensorSpecifications;
+  {
+    auto pinholeCameraSpec = esp::sensor::CameraSensorSpec::create();
+    pinholeCameraSpec->sensorSubType = esp::sensor::SensorSubType::Pinhole;
+    pinholeCameraSpec->sensorType = esp::sensor::SensorType::Color;
+    pinholeCameraSpec->position = {0.0f, 0.f, 0.0f};
+    pinholeCameraSpec->resolution = {512, 384};
+    pinholeCameraSpec->uuid = sensorName;
+    sensorSpecifications = {pinholeCameraSpec};
+  }
+
+  ReplayBatchRendererConfiguration batchRendererConfig;
+  batchRendererConfig.sensorSpecifications = std::move(sensorSpecifications);
+  batchRendererConfig.numEnvironments = numEnvs;
+  ReplayBatchRenderer batchRenderer(batchRendererConfig);
+  auto renderer = batchRenderer.getRenderer();
+
+  std::vector<std::vector<char>> buffers(numEnvs);
+  std::vector<Mn::MutableImageView2D> imageViews;
```

Comment:
FYI you could use Magnum's [Image2D](https://doc.magnum.graphics/magnum/classMagnum_1_1Image.html) instead of these two. It's an owning container so there's no need to maintain a separate buffer for it, and it's implicitly convertible to a `[mutable]ImageView2D` as well.

Reply:
I'll move on with this changeset and improve the test case when the real batch renderer will be integrated here.